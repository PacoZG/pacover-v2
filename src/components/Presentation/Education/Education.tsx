import React from 'react'
import { useTranslations } from 'next-intl'
import '../education-style.css'

const Education: React.FC = () => {
  const t = useTranslations('education')

  return (
    <div className="flex flex-col items-center p-4 md:py-10 lg:py-20">
      <div className="flex flex-col items-center bg-gray-600 dark:bg-gray-900 w-full sm:w-11/12 lg:w-2/3 transition duration-500 rounded-xl bg-opacity-70 p-4">
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://devopswithkubernetes.com/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line12')}
            </a>
          </h1>

          <p className="education-card-description">{t('line11')}</p>
        </div>

        {/* Education Entry 2: DevOps with Docker */}
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://devopswithdocker.com/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line10')}
            </a>
          </h1>

          <p className="education-card-description">{t('line9')}</p>
        </div>

        {/* Education Entry 3: Full Stack Open */}
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://fullstackopen.com/en/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line8')}
            </a>
          </h1>

          <p className="education-card-description">{t('line7')}</p>
        </div>

        {/* Education Entry 4: Aalto University */}
        <div className="education-card">
          <h1 className="education-card-title">{t('line6')}</h1>

          <p className="education-card-description">{t('line5')}</p>

          <a
            education-tool-tip-msg={t('tool-tip-msg')}
            className="education-link-secondary"
            href="https://www.aalto.fi/en/school-of-electrical-engineering"
            target="_blank"
            rel="noopener noreferrer"
          >
            {t('line4')}
          </a>
        </div>

        {/* Education Entry 5: Technical University of Cartagena */}
        <div className="education-card">
          <h1 className="education-card-title">{t('line3')}</h1>
          <p className="education-card-description">{t('line2')}</p>
          <a
            education-tool-tip-msg={t('tool-tip-msg')}
            className="education-link-secondary"
            href="http://www.upct.es"
            target="_blank"
            rel="noopener noreferrer"
          >
            {t('line1')}
          </a>
        </div>
      </div>
    </div>
  )
}

export default Education
