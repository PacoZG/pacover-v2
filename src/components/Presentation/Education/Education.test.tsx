import React from 'react'
import { render } from '@testing-library/react'
import { describe, test, expect, vi, beforeEach } from 'vitest'
import Education from '@/components/Presentation/Education/Education'

describe('Health', () => {
  describe('given configured with dependencies', () => {
    beforeEach(() => {
      window.scrollTo = vi.fn()

      vi.mock('next-intl', () => ({
        useTranslations: () => (key: string) => key,
      }))
    })

    describe('Education component', () => {
      test('matches the snapshot', () => {
        const { container } = render(<Education />)

        expect(container).toMatchSnapshot()
      })
    })
  })
})
