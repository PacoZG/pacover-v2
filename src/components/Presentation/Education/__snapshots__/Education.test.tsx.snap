// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Health > given configured with dependencies > Education component > matches the snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center p-4 md:py-10 lg:py-20"
  >
    <div
      class="flex flex-col items-center bg-gray-600 dark:bg-gray-900 w-full sm:w-11/12 lg:w-2/3 transition duration-500 rounded-xl bg-opacity-70 p-4"
    >
      <div
        class="education-card"
      >
        <h1
          class="education-card-title"
        >
          <a
            class="education-link"
            href="https://devopswithkubernetes.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            line12
          </a>
        </h1>
        <p
          class="education-card-description"
        >
          line11
        </p>
      </div>
      <div
        class="education-card"
      >
        <h1
          class="education-card-title"
        >
          <a
            class="education-link"
            href="https://devopswithdocker.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            line10
          </a>
        </h1>
        <p
          class="education-card-description"
        >
          line9
        </p>
      </div>
      <div
        class="education-card"
      >
        <h1
          class="education-card-title"
        >
          <a
            class="education-link"
            href="https://fullstackopen.com/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            line8
          </a>
        </h1>
        <p
          class="education-card-description"
        >
          line7
        </p>
      </div>
      <div
        class="education-card"
      >
        <h1
          class="education-card-title"
        >
          line6
        </h1>
        <p
          class="education-card-description"
        >
          line5
        </p>
        <a
          class="education-link-secondary"
          education-tool-tip-msg="tool-tip-msg"
          href="https://www.aalto.fi/en/school-of-electrical-engineering"
          rel="noopener noreferrer"
          target="_blank"
        >
          line4
        </a>
      </div>
      <div
        class="education-card"
      >
        <h1
          class="education-card-title"
        >
          line3
        </h1>
        <p
          class="education-card-description"
        >
          line2
        </p>
        <a
          class="education-link-secondary"
          education-tool-tip-msg="tool-tip-msg"
          href="http://www.upct.es"
          rel="noopener noreferrer"
          target="_blank"
        >
          line1
        </a>
      </div>
    </div>
  </div>
</div>
`;
